src/librbd/io/CopyupRequest.cc

```diff
@@ -233,6 +234,7 @@ void CopyupRequest<I>::handle_deep_copy(int r) {
 
   m_image_ctx->snap_lock.get_read();
   m_lock.Lock();
+  m_copyup_is_zero = m_copyup_data.is_zero();
```

Comment:
Nit: the ```m_copyup_data``` will be empty here after a deep-copy, so ```m_copyup_is_zero``` can just be hard-coded to true.